### DISCUSSION TITLE
TODO ideas for future releases

### DISCUSSION THREAD

**Discussion by Aurelien30000** [IntellectualSites/fastasyncvoxelsniper#52] [Ideas]
Thank you again and again for precise review, pr is almost ready imo, **needs some further testing though, brush by brush** (I'll do it on my side on Saturday).

_No more changes due to lack of time to devote to FAVS, huge inconsistencies with for instance spiral stair case brush will be addressed later, I will try for the end of August._

_Originally posted by @Aurelien30000 in https://github.com/IntellectualSites/FastAsyncVoxelSniper/issues/50#issuecomment-897994254_

**Comment by NotMyFault** [comment]
A few other things that may be handy:

- Tab completion for basic commands such as `/vs toggle, info` etc. including permission: https://github.com/IntellectualSites/FastAsyncVoxelSniper/pull/57
- The ability to use brushes outside of Y 0 -> 256: https://github.com/IntellectualSites/FastAsyncVoxelSniper/pull/57
- More descriptive no permission messages, e.g if the player has no permission to use the sniper in an area it'll tell you that the snipe block must be visible over actively notifying about missing permission.
- Utilize `LivingEntity#addPotionEffects` in PunishBrush, multiple effects of the same type are now supported.: https://github.com/IntellectualSites/FastAsyncVoxelSniper/pull/57
- Replace local variables in favor of command arguments or config keys, whatever applies in a scenario better; see `BRUSH_SIZE_WARNING_THRESHOLD` for example.: https://github.com/IntellectualSites/FastAsyncVoxelSniper/pull/57
- Don't tab complete brushes to you have no permission to use: https://github.com/IntellectualSites/FastAsyncVoxelSniper/pull/57

Very very further in the future:
- Replace legacy colors codes (`ChatColor.<key>`) with more future proof concepts of message handling, e.g. [MiniMessage](https://github.com/KyoriPowered/adventure-text-minimessage). That offers the ability to style messages with a semi-standardized pattern while being able to translate messages too.

**Reply by Aurelien30000** [reply]
Great ideas and improvements!


> The ability to use brushes outside of Y 0 -> 256

_Already made required changes hopefully, using EditSession methods, waiting for FAWE compatibility atm._

**Reply by NotMyFault** [reply]
I tried to use it earlier with the pr that adds it to Fawe but I wasn't successful doing so

**Reply by Aurelien30000** [reply]
Seems good overall, per-haps some hardcoded values must be changed.
Since all brushes are configured to use common methods that directly interacts with FAWE, FAVS works with custom heights ;) 

https://streamable.com/vfa8gm

**Reply by NotMyFault** [reply]
https://streamable.com/wve5yz

**Reply by Aurelien30000** [reply]
nice catch 😓

**Reply by Aurelien30000** [reply]
> More descriptive no permission messages, e.g if the player has no permission to use the sniper in an area it'll tell you that the snipe block must be visible over actively notifying about missing permission.

Wdym exactly here, no error message is sent by VS iirc, and other ones are usually interaction protection 🤔?

**Reply by NotMyFault** [reply]
I guess that resolved it self? Previously when you attempted to use FAVS in a region you have no permission to, it told you the snipe target block must be visible rather than you have no permission to use FAVS here. Atm I'm not able to reproduce it, but I guess you get it.

**Comment by Aurelien30000** [comment]
**Idea from @JayemCeekay:** For some brushes like ``erode``, I would be in favor of leaving the choice, an individual command per arg, as well as a general command for all args.

_- Should be addressed when switching to cloud._

**Comment by Aurelien30000** [comment]
**Idea from @NotMyFault**: How does one apply several punishments? Like /b punish weakness,fire is what I thought of.

_- Should add a way to apply several punishments at the same time._

**Reply by NotMyFault** [reply]
It's unlikely someone ever needs to punish their players or mobs in such a bad way, but it's possible and we could do so with a comma separated list e.g.

**Comment by Aurelien30000** [comment]
**Idea from @NotMyFault:** Is this how the brush is supposed to look like? yo 5 & yl 3
Looks alike the canyon brush just in weird?
Likely, removing the entire chunk is eh.. weird to place a few rows of water and dirt. 

_- I imagined it would act like any other brush and putting down the blocks on top of the block you target on, instead of erasing a whole chunk._
_- Create an option to specify material instead of classic dirt._

**Comment by Aurelien30000** [comment]
**Idea from @Aurelien30000:** Rewrite ``SpiralStaircaseBrush``.

_- Add an arg to pass the width of the staircase. Voxel height is the default implemented arg for several brushes._
_- Accept all stairs and steps._

**Comment by NotMyFault** [comment]
In order to comply with the requirements of the central repository, artifacts must use a group id you can verify you own.
I was considering `com.intellectualsites` or something more specific: https://github.com/IntellectualSites/FastAsyncVoxelSniper/pull/58
